engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/ClassMetaData.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.metadata.aggregated;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.validation.ElementKind;
+
+import org.hibernate.validator.internal.engine.valueextraction.ValueExtractorManager;
+import org.hibernate.validator.internal.metadata.core.ConstraintHelper;
+import org.hibernate.validator.internal.metadata.core.MetaConstraint;
+import org.hibernate.validator.internal.metadata.descriptor.ClassDescriptorImpl;
+import org.hibernate.validator.internal.metadata.raw.ConstrainedElement;
+import org.hibernate.validator.internal.metadata.raw.ConstrainedType;
+import org.hibernate.validator.internal.util.TypeResolutionHelper;
+
+/**
+ * Represents the constraint related meta data for a type i.e. class-level
+ * constraints.
+ *
+ * @author Gunnar Morling
+ * @author Guillaume Smet
+ * @author Marko Bekhta
+ */
+public class ClassMetaData extends AbstractConstraintMetaData {
+
+	private ClassMetaData(Class<?> beanClass,
+			Set<MetaConstraint<?>> constraints,
+			Set<MetaConstraint<?>> containerElementsConstraints) {
+		super(
+				beanClass.getSimpleName(),
+				beanClass,
+				constraints,
+				containerElementsConstraints,
+				false,
+				!constraints.isEmpty() || !containerElementsConstraints.isEmpty()
+		);
+	}
+
+	@Override
+	public ClassDescriptorImpl asDescriptor(boolean defaultGroupSequenceRedefined, List<Class<?>> defaultGroupSequence) {
+		return new ClassDescriptorImpl(
+				getType(),
+				asDescriptors( getDirectConstraints() ),
+				defaultGroupSequenceRedefined,
+				defaultGroupSequence
+		);
+	}
+
+	@Override
+	public String toString() {
+		return "ClassLevelMetaData [type=" + getType() + "]]";
```

Comment:
s/ClassLevelMetaData/ClassMetaData/

Reply:
@marko-bekhta see, that's why I want us to use `getClass().getSimpleName()` :)

Reply:
And I trusted IDEA to be good with such kind of refactorings!!! :smile: Thanks!